src/main/lib/osu-file-parser/OsuParser.ts

```diff
@@ -1,125 +1,345 @@
-        }
+    const db_version = db.read_u32();
+    console.debug("db version:", db_version);
+    if (db_version < 20170222) {
+      return fail("osu!.db is too old, please update the game.");
+    }
 
-        const audioID = path.join(dirs[i], audioSrc.value);
-        let s: Song | undefined = undefined;
+    db.read_u32(); // folder count
+    db.read_u8();
+    db.read_u64(); // timestamp
 
-        if (audioID === lastAudioID) {
-          s = lastAudioSong;
-        } else {
-          const a = audioTable.get(audioID);
+    const player_name = db.read_string().trim();
+    console.debug("player name:", player_name);
 
-          if (a !== undefined) {
-            s = songTable.get(a.songID);
-          }
+    const nb_beatmaps = db.read_u32();
+    console.debug("nb beatmaps:", nb_beatmaps);
 
-          if (s !== undefined) {
-            lastAudioID = audioID;
-            lastAudioSong = s;
-          }
+    let last_audio_filepath = "";
+    for (let i = 0; i < nb_beatmaps; i++) {
+      try {
+        if (db_version < 20191107) {
+          // https://osu.ppy.sh/home/changelog/stable40/20191107.2
+          db.read_u32();
         }
 
-        if (s !== undefined) {
-          const diff = await OsuFile.getProp(file, "Version");
+        const song: Song = {
+          audio: "",
+          osuFile: "",
+          path: "",
+          ctime: "",
+          dateAdded: "",
+          title: "",
+          artist: "",
+          creator: "",
+          bpm: [],
+          duration: 0,
+          diffs: []
+        };
+
+        song.artist = db.read_string().trim();
+        song.artistUnicode = db.read_string().trim();
+        song.title = db.read_string().trim();
+        song.titleUnicode = db.read_string().trim();
+        song.creator = db.read_string().trim();
+
+        // NOTE: I'm being lazy here, and only loading the first diff for a given audio file.
+        //       This is wrong, especially if diffs of a set aren't right next to each other in the db.
+        const diff_name = db.read_string();
+        song.diffs = [diff_name];
+
+        const audio_filename = db.read_string();
+
+        db.read_hash(); // .osu file md5
+        const osu_filename = db.read_string().trim();
+        db.read_u8(); // ranking status
+        db.read_u16(); // nb circles
+        db.read_u16(); // nb sliders
+        db.read_u16(); // nb spinners
+
+        // TODO: This probably is incorrect. See alternatives:
+        // tms_a = (tms - 621355968000000000) / 10000000
+        // tms_b = (tms - 504911232000000000)
+        const last_modification_time = db.read_u64();
+        song.dateAdded = new Date(last_modification_time).toISOString();
+
+        db.read_f32(); // AR
+        db.read_f32(); // CS
+        db.read_f32(); // HP
+        db.read_f32(); // OD
+        db.read_f64(); // slider multiplier
+
+        // std
+        let nb_star_ratings = db.read_u32();
+        for (let s = 0; s < nb_star_ratings; s++) {
```

Comment:
Seems like this piece of code just skips `N` bytes. This should be converted to 
```javascript
reader.skip(N) // Optional description of jump
```

Reply:
fixed
reference: https://discord.com/channels/1284644086820896879/1289979431145701498/1292183512434348074